libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -134,6 +134,11 @@ func (p *pod) OnAdd(obj interface{}) {
 func (p *pod) OnUpdate(obj interface{}) {
 	pod := obj.(*kubernetes.Pod)
 
+	p.logger.Debugf("Watcher Pod update for pod: %+v, status: %+v", pod.Name, pod.Status.Phase)
+	if pod.Status.Phase == kubernetes.PodPending {
```

Comment:
We have now two consecutive ifs based on `pod.Status.Phase`, maybe they could be replaced with a `switch`.